quixstreams/app.py

```diff
@@ -197,27 +198,32 @@ def __init__(
             "Quix__Consumer_Group", "quixstreams-default"
         )
 
-        if quix_config_builder:
-            quix_app_source = "Quix Config Builder"
-        if quix_config_builder and quix_sdk_token:
-            raise warnings.warn(
-                "'quix_config_builder' is not necessary when an SDK token is defined; "
-                "we recommend letting the Application generate it automatically"
-            )
-
-        if quix_sdk_token and not quix_config_builder:
-            quix_app_source = "Quix SDK Token"
-            quix_config_builder = QuixKafkaConfigsBuilder(quix_sdk_token=quix_sdk_token)
+        if broker_address:
+            # If broker_address is passed to the app it takes priority over any quix configuration
+            self._is_quix_app = False
+            topic_manager_factory = TopicManager
+            if isinstance(broker_address, str):
+                broker_address = ConnectionConfig(bootstrap_servers=broker_address)
+        else:
+            self._is_quix_app = True
+            if quix_config_builder:
+                quix_app_source = "Quix Config Builder"
+            if quix_config_builder and quix_sdk_token:
+                raise warnings.warn(
+                    "'quix_config_builder' is not necessary when an SDK token is defined; "
+                    "we recommend letting the Application generate it automatically"
+                )
 
-        if broker_address and quix_config_builder:
-            raise ValueError(
-                'Cannot provide both "broker_address" and "quix_sdk_token"'
-            )
-        elif not (broker_address or quix_config_builder):
-            raise ValueError(
-                'Either "broker_address" or "quix_sdk_token" must be provided'
-            )
-        elif quix_config_builder:
+            if not quix_config_builder:
+                if quix_sdk_token:
+                    quix_app_source = "Quix SDK Token"
+                    quix_config_builder = QuixKafkaConfigsBuilder(
+                        quix_sdk_token=quix_sdk_token
+                    )
+                else:
+                    raise ValueError(
+                        'Either "broker_address" or "quix_sdk_token" must be provided'
+                    )
```

Comment:
I like the approach overall, i.e., for the `broker_address` to come before `quix_sdk_token` if it's defined.

On the coding side, can we reduce the number of the `if` clauses here and make the conditions more straightforward?